### PR TITLE
Azure Monitor Exporter - Expand extension methods for OpenTelemetry dependency injection support

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
- Added default constructor to `AzureMonitorExporterOptions` so that it could be used as template. Parameterless constructor is needed to invoke `options`
- Modified the extension methods for `AddAzureMonitorTraceExporter` and `AddAzureMonitorMetricExporter`
Configuration providers that are added later override previous key settings. 

If `AzureMonitorExporterOptions` are set in both code and `IConfiguration` (e.g., appSettings.json). Value from the `IConfiguration` takes precedence over the code.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterMetricExtensions.cs [hunk 1]
```diff
@@ -28,11 +28,26 @@ public static MeterProviderBuilder AddAzureMonitorMetricExporter(this MeterProvi
                 throw new ArgumentNullException(nameof(builder));
             }
 
-            var options = new AzureMonitorExporterOptions();
-            configure?.Invoke(options);
+            if (builder is IDeferredMeterProviderBuilder deferredMeterProviderBuilder)
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterMetricExtensions.cs:31]: Can we follow up with @CodeBlanch about this?
I was under the impression that OTel was moving away from the IDeferredProviderBuilder.

See also: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.InMemory/InMemoryExporterMetricsExtensions.cs

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterMetricExtensions.cs:31]: I see, will get this fixed. Will follow the recommendation from here - https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/extending-the-sdk#tracerproviderbuilder-extension-methods

Comment (CodeBlanch) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterMetricExtensions.cs:31]: So what is on this PR will work with the 1.3 API and continue to work with the 1.4 API but doesn't get you all the way there. It is also very ugly 🤣 The 1.4 API allows for a deeper more standard options integration without reflection in a little cleaner style.

Can merge this and update to the 1.4 API later or wait for it either works for me.

**Review by vishweshbankwar — APPROVED** [review]
LGTM: Would be good to add a comment regarding which setting will take preference i.e. via code or via appsettings.

We will merge after the release is completed today.

**Comment by TimothyMothra** [comment]
> Would be good to add a comment regarding which setting will take preference i.e. via code or via appsettings.

This would be good to document as it comes up in support questions.
May be time to revisit the Readme and confirm it's up to date with config options

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterTraceExtensions.cs [hunk 2]
```diff
@@ -16,27 +18,49 @@ namespace Azure.Monitor.OpenTelemetry.Exporter
...
+                    // configuration delegate inline. The reason for this is
+                    // AzureMonitorExporterOptions is shared by all signals. Without a
+                    // name, delegates for all signals will mix together. See:
+                    // https://github.com/open-telemetry/opentelemetry-dotnet/issues/4043
+                    configure(exporterOptions);
+                }
 
-            // TODO: Pick Simple vs Batching based on AzureMonitorExporterOptions
-            return builder.AddProcessor(new BatchActivityExportProcessor(new AzureMonitorTraceExporter(options, credential)));
+                return new BatchActivityExportProcessor(new AzureMonitorTraceExporter(exporterOptions, credential));
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterTraceExtensions.cs:62]: do we allow changing BatchExport options?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterTraceExtensions.cs:62]: There are no plans to change BatchExport options.

**Pull request merged**